backends/pipeline_api_test.py

```diff
@@ -166,3 +167,33 @@ def test_update_profile(self, mocked_get_json):
         ]
 
         self.check_profile_fields(self.user_profile, all_fields)
+
+
+class LinkedInPipelineTests(TestCase):
+    """Tests for the LinkedIn pipline entry"""
+
+    def test_saves_linkedin_response(self):
+        """Happy path"""
+        backend = mock.Mock()
+        backend.name = 'linkedin-oauth2'
+        user = UserFactory.create()
+        response = {'test': 'works'}
+        is_new = True
+        update_from_linkedin(backend, user, response, is_new)
+
+        profile = Profile.objects.get(user=user)
+        assert profile.linkedin == {'test': 'works'}
+
+    def test_doesnt_update_on_old_entries(self):
+        """Shouldn't update entry if it's already fetched"""
+        backend = mock.Mock()
+        backend.name = 'linkedin-oauth2'
+        user = UserFactory.create()
+        response = {'test': 'works'}
+        is_new = True
+        update_from_linkedin(backend, user, response, is_new)
+        is_new = False
+        update_from_linkedin(backend, user, {'doesnt': 'work'}, is_new)
+
+        profile = Profile.objects.get(user=user)
+        assert 'doesnt' not in profile.linkedin
```

Comment:
Maybe it is too much, but what if the profile does not exist (after all there is a check in the code)